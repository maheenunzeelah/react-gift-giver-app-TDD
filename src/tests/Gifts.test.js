import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import Gift from '../components/gifts';

Enzyme.configure({adapter:new Adapter()});
const id=1
const mockFunction=jest.fn();
const props={gift:{id},removeGift:mockFunction}
const gift=shallow(<Gift {...props}/>);
describe("Gifts", ()=>{
 it("renders correctly",()=>{
  expect(toJSON(gift)).toMatchSnapshot()
 })

  it("Initial state of person and present state",()=>{
      expect(gift.state()).toEqual({person:'',present:''})
  })
  describe("When typing into person input",()=>{
      const person='Uncle';
      beforeEach(()=>{
          gift.find('.input-person').simulate('change',{target:{value:person}});
      })
      it("updates the person in state",()=>{
          expect(gift.state().person).toEqual(person)
      })
  });
  describe("When typing into present input",()=>{
    const present='Watch';
    beforeEach(()=>{
        gift.find('.input-present').simulate('change',{target:{value:present}});
    })
    it("updates the present in state",()=>{
        expect(gift.state().present).toEqual(present)
    }) 
  })
  describe("When clicking on `removeGift` button",()=>{
      beforeEach(()=>{
          gift.find('.btn-remove').simulate('click');
      })
      it("button calls a callback function",()=>{
       expect(mockFunction).toHaveBeenCalledWith(id)
      })
  })
});