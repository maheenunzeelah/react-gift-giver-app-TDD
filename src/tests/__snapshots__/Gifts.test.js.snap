// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gifts renders correctly 1`] = `
<div
  className="container"
>
  <Form
    inline={false}
  >
    <FormGroup
      controlId="formPlaintext"
    >
      <FormLabel
        column={false}
        srOnly={false}
      >
        Person
      </FormLabel>
      <FormControl
        className="input-person"
        onChange={[Function]}
        placeholder="Person Name"
        size="sm"
        type="text"
      />
      <br />
      <FormLabel
        column={false}
        srOnly={false}
      >
        Present
      </FormLabel>
      <FormControl
        className="input-present"
        onChange={[Function]}
        placeholder="Present Name"
        size="lg"
        type="text"
      />
    </FormGroup>
    <Button
      active={false}
      className="btn-remove"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Remove Gift
    </Button>
  </Form>
</div>
`;
